rpc-rs/src/otel.rs

```diff
@@ -115,9 +118,17 @@ impl From<OtelHeaderInjector> for HeaderMap {
 /// A convenience function that will extract the current context from NATS message headers and set
 /// the parent span for the current tracing Span. If you want to do something more advanced, use the
 /// [`OtelHeaderExtractor`] type directly
-pub fn attach_span_context(msg: &crate::async_nats::Message) {
-    let header_map = OtelHeaderExtractor::new_from_message(msg);
+pub fn attach_span_context(traceparent: &Option<&str>, tracestate: &Option<&str>) {
+    let mut header_map = HeaderMap::new();
+    if let Some(p) = traceparent {
+        header_map.insert(HEADER_TRACEPARENT, *p);
+    }
+    if let Some(p) = tracestate {
+        header_map.insert(HEADER_TRACESTATE, *p);
+    }
+
+    let extractor = OtelHeaderExtractor::new(&header_map);
```

Comment:
I think we should keep around OtelHeaderExtractor for when we get to 1.0, but we should probably create a separate extractor that doesn't use a header map so we aren't allocating anything. You can probably copy paste the current extractor code and then have it be something like 

```rust
struct RawDataExtractor<'a> {
    traceparent: &'a Option<&'a str>,
    tracestate: &'a Option<&'a str>,
}

impl<'a> Extractor for RawDataExtractor<'a> {
    fn get(&self, key: &str) -> Option<&str> {
       match key {
           HEADER_TRACEPARENT => self.traceparent,
           HEADER_TRACESTATE => self.tracestate,
           _ => None
        }
    }
    fn keys(&self) -> Vec<&str> {
        vec![HEADER_TRACEPARENT, HEADER_TRACESTATE]
    }
}
```

That should make it so there is no allocation happening at all

Reply:
- [ ] refactor to avoid extra alloc